core/sr-primitives/src/generic/mod.rs

```diff
@@ -42,8 +42,8 @@ use rstd::prelude::*;
 fn encode_with_vec_prefix<T: Encode, F: Fn(&mut Vec<u8>)>(encoder: F) -> Vec<u8> {
 	let size = ::rstd::mem::size_of::<T>();
 	let reserve = match size {
-		0..=0b00111111 => 1,
-		0..=0b00111111_11111111 => 2,
+		0..=0b0011_1111 => 1,
+		0b0100_0000..=0b0011_1111_1111_1111 => 2,
```

Comment:
I feel like there is a faster way to do this using some obscure CPU instruction, but I do not believe the effort is worth it.